packages/panzer/disc-fe/src/evaluators/Panzer_Integrator_GradBasisDotVector_impl.hpp

```diff
@@ -278,13 +270,13 @@ namespace panzer
       for (int qp(0); qp < numQP; ++qp)
       {
 	team.team_barrier();
-	tmp_(cell) = 1.0;
+        ScalarT fieldMultsTotal(1);
```

Comment:
This will not work for AMD machines. If ScalarT is a dynamic fad type, it tries to allocate the derivative array on device while the kernel is running. This capability is not supported by AMD. We refactored all evaluators to not use "ScalarT tmp" but to allocate the views for tmp_ array before the kernel is launched.

Reply:
Hm. Guess I'll have to revisit that. I changed it and it circumvented some errors on Cuda.

Reply:
That's weird. I'm routinely building the panzer test suite with uvm off for empire work and not seeing any failures. What kind of failures were you seeing?

Reply:
There might be some testing gaps? Otherwise I don't understand how `Integrator_GradBasisDotVector::postRegistrationSetup` worked without UVM.

Reply:
Is it possible that there is no testing with "Field Multipliers"? That would explain how `postRegistrationSetup` worked, and it could also explain how I was seeing issues in `operator()`.

Reply:
That is possible. Let me take a look at the tests...

Reply:
@rppawlo I reverted the change to the `tmp_` array, and the matrix values change compared to a serial run (and the solver fails to converge). With the change, I only saw differences at the order of machine precision.

Reply:
@rppawlo  I built on HIP. With the change to `Integrator_GradBasisDotVector` the runs fail with an allocation error. I do get past that without the change, but the symptoms are like on Cuda. Now, it's obviously possible that something else in my PR breaks things in a non-trivial way. Is there a test in Panzer that I could easily extend with field multipliers?

Reply:
@cgcgcg - there is a simple energy equation example in:
```
Trilinos/packages/panzer/adapters-stk/example/main_driver
```
To add the multiplier, you will need to edit the file:
```
Trilinos/packages/panzer/disc-fe/test/equation_set/user_app_EquationSet_Energy_impl.hpp
```
to add in the multiplier when GradBasisDotVector is constructed for the diffusion term. There are three paths through the kernel - no multipliers, one multiplier and more than one multiplier. We should probably add tests for all three paths at some point.

Reply:
@rppawlo I _think_ I got it now. After looking at some of the other evaluators, it seems that the `kokkosFieldMultipliers` inner view should be unmanaged. As far as I can tell, this change fixed my issues on Cuda. I'll rebuild on HIP as well to double check the changes of this PR.

Reply:
Cool! Yes - inner views must be unmanaged. To make this easier to remember (and handle memory management issues), there are actually three implementations of ViewOfView objects for managing different use cases. Since you have it working, I would not change anything, but in the future, would recommend using the ViewOfViews3 object (see Phalanx_KokkosViewOfViews.hpp) as it makes the inner views unmanaged at runtime by using the pointer constructor - the template parameter is then not needed on the inner views. This works best for backwards compatibility.

Reply:
The Panzer tests are all passing on HIP (minus some that failed since the machine has NetCDF without parallel support). I will split out the Tpetra/Ifpack2/MueLu changes to reduce this PR to Panzer only changes.